hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,344 @@
+import numpy as np
+import matplotlib.pyplot as plt 
+from collections import namedtuple
+
+from hnn_core import read_params, jones_2009_model, simulate_dipole 
+from hnn_core.network import pick_connection
+
+from skopt import gp_minimize 
+from skopt.utils import use_named_args
+from skopt.space import Real
+
+from scipy.optimize import fmin_cobyla
+
+class Optimizer:
+    def __init__(self, net, constraints, solver, metric):
+        self.net = net
+        self.constraints = constraints
+        if solver == 'bayesian':
+            self._get_params = _get_params_bayesian
+            self._run_opt = _run_opt_bayesian
+        elif solver == 'cobyla':
+            self._get_params = _get_params_cobyla
+            self._run_opt = _run_opt_cobyla
+        if metric == 'evoked':
+            self.metric = _rmse_evoked
+        elif metric == 'rhythmic':
+            self.metric = _rmse_rhythmic
```

Comment:
I like the `metric` argument here, but how about `self.obj_func` instead of `self.metric` so that it's clear this translates to a function?